src/components/theme/View/LinkView.jsx

```diff
@@ -1,92 +1,63 @@
-/**
- * Link View.
- * @module components/theme/View/LinkView
- */
-
-import React, { Component } from 'react';
+import { useEffect } from 'react';
 import PropTypes from 'prop-types';
 import { isInternalURL, flattenToAppURL } from '@plone/volto/helpers';
 import { Container } from 'semantic-ui-react';
 import { UniversalLink } from '@plone/volto/components';
 import { FormattedMessage } from 'react-intl';
 import config from '@plone/volto/registry';
 
-/**
- * View container class.
- * @class View
- * @extends Component
- */
-class LinkView extends Component {
-  /**
-   * Property types.
-   * @property {Object} propTypes Property types.
-   * @static
-   */
-  static propTypes = {
-    content: PropTypes.shape({
-      title: PropTypes.string,
-      description: PropTypes.string,
-      remoteUrl: PropTypes.string,
-    }),
-    token: PropTypes.string,
-  };
-
-  /**
-   * Default properties.
-   * @property {Object} defaultProps Default properties.
-   * @static
-   */
-  static defaultProps = {
-    content: null,
-    token: null,
-  };
-
-  componentDidMount() {
-    if (!this.props.token) {
-      const { remoteUrl } = this.props.content;
+const LinkView = ({ token, content, history }) => {
+  useEffect(() => {
+    if (!token) {
+      const { remoteUrl } = content;
       if (isInternalURL(remoteUrl)) {
-        this.props.history.replace(flattenToAppURL(remoteUrl));
+        history.replace(flattenToAppURL(remoteUrl));
       } else if (!__SERVER__) {
         window.location.href = flattenToAppURL(remoteUrl);
       }
     }
-  }
+  });
```

Comment:
This useEffect will fire on all re-renders. You can pass an empty dependency array to prevent that.